config/application.rb

```diff
@@ -89,53 +86,6 @@ class Application < Rails::Application
       end
     end
 
-    initializer "user_authentication" do |app|
-      Decidim::User.send(:include, UserAuthentication)
-
-      # The following hook is for the development environment and it is needed
-      # to load the correct omniauth configurations to the Decidim::User model
-      # BEFORE the routes are reloaded in Decidim::Core::Engine. Without this,
-      # the extra omniauth authentication methods are lost during application
-      # reloads as the Decidim::User class is reloaded during which the omniauth
-      # configurations are overridden by the core class. After the override, the
-      # routes are reloaded (before call to to_prepare) which causes the extra
-      # configured methods to be lost.
-      #
-      # The load order is:
-      # - Models, including Decidim::Core::Engine models (sets the omniauth back
-      #   to Decidim defaults)
-      # - ActionDispatch::Reloader - after_class_unload hook (below)
-      # - Routes, including Decidim::Core::Engine routes (reloads the routes
-      #   using the omniauth providers set by Decidim::Core)
-      # - to_prepare hook (which would be the optimal place for this but too
-      #   late in the load process)
-      #
-      # In case you are planning to change this, make sure that the following
-      # works:
-      # - Start the application with Tunnistamo omniauth method configured
-      # - Load the login page and see that Tunnistamo is configured
-      # - Make a change to any file under the `app` folder
-      # - Reload the login page and see that Tunnistamo is configured
-      #
-      # This could be also fixed in the Decidim core by making the omniauth
-      # providers configurable through the application configs. See:
-      # https://github.com/decidim/decidim/blob/a40656/decidim-core/app/models/decidim/user.rb#L17
-      #
-      # NOTE: This problem only occurs when the models and routes are reloaded,
-      #       i.e. in development environment.
-      app.reloader.after_class_unload do
-        Decidim::User.send(:include, UserAuthentication)
-      end
-    end
-
-    initializer "devise_overrides" do
```

Comment:
I'm not sure this block should be removed. I think it's unrelated to the above block

Reply:
You're right, this appears to be a Helsinki add but not one with negative side effects. Re-added.

Reply:
I think if we wanted to drop this block (I don't really know the intent of the Helsinki add) then we would just want to delete the `DeviseOverrides` class as well. Feel free to do what you think is best :)

Reply:
There's a comment in the DeviseOverrides class saying [this PR](https://github.com/decidim/decidim/pull/5398) (which has been merged) would make this unnecessary. Will strip them both out, the less code the better!